src/ScrollToHOC.jsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import getDisplayName from "react-display-name";
+import handleScroll from "./handleScroll";
+
+/**
+ * Higher Order Component version of the ScrollTo.
+ * Injects a prop named scroll that is a function that
+ * takes an (x, y) coordinate to scroll to. [ie. props.scroll(0, 500)]
+ */
+function ScrollToHOC(Component) {
+  const WrappedComponent = props => (
+    <Component {...props} scroll={handleScroll} />
```

Comment:
Doesn't need to be solved in this PR, but it'd be cool to also use `ScrollArea` inside a HOC.

```
const Elem = ScrollToHOC((props) => (
   <ScrollArea>
      <button onClick={props.scroll(0, 200)}>Click</button>
   </ScrollArea>
));
```

I'll continue the conversation in #15!

Reply:
Sounds good. I added a comment to #15.

I'm okay with adding the `ScrollArea` functionality for `HOC` in a different Pull Request.